Release `1.0.0` : décollage vers les astres 🚀

goulvenclech opened the pull request:
On embarque dans la fusée d'[Astro](https://astro.build/) pour se débarrasser de Vue, Gridsome, Yarn, et de tous les poids morts qui encombraient la béta !

goulvenclech committed:
Ignition with astro 🚀

goulvenclech committed:
refactor(routing): wip dynamic page routing & wip Courses
Severe lack of typing tho.

Princesseuh commented on src/layouts/ContentLayout.astro:
```diff
@@ -0,0 +1,16 @@
+import type { Course } from "../data/courses"
+import AppLayout from "./AppLayout.astro"
+
+const { content } = Astro.props
```
Tu peux typer `Astro.props` directement en mettant une interface / type nommé `Props` dans le fichier: https://docs.astro.build/en/guides/typescript/#component-props

Princesseuh commented on src/pages/[course].astro:
```diff
@@ -0,0 +1,21 @@
+import { getSlugFromPath } from "../utils"
+
+export async function getStaticPaths() {
+  const courses = await Astro.glob("../../content/**/index.md")
```
ah en fait, j'remarque tu le fais dans un autre fichier, donc tu dois déjà savoir, dsl

Princesseuh commented on src/utils/index.ts:
```diff
@@ -0,0 +1,20 @@
+ * @param contentType what's the type of content ?
+ * @returns
+ */
+export function getSlugFromPath(path: String, contentType: String): String {
```
pour le deuxième, tu peux utiliser un union plutôt qu'un raw string `contentType: 'course' | 'whatever'` aussi. Ça rendrait la switch plus rapide à écrire

GoulvenC commented on src/utils/index.ts:
```diff
@@ -0,0 +1,20 @@
+export function getSlugFromPath(path: String, contentType: String): String {
+  switch (contentType) {
+    case "course":
+      return path.substring(
+        // there is probably a smarter way
+        path.indexOf("content/") + 8,
+        path.indexOf("/index.md")
+      )
```
Meh. Ma version est plus stupide, mais plus précise et solide. J'y penserais plus tard.

goulvenclech committed:
fix(courses): better typing

goulvenclech committed:
feat(routing): wip getStaticPathFromContentType & wip Chapter
getStaticPathFromContentType add a standardized way to get static path.
But the function throw an obscure error when used in TS files,
See this issue -> https://github.com/withastro/astro/issues/5552

Princesseuh committed:
update(config): Updated formatting config

aureliendossantos committed:
refactor(content): markdown to mdx

goulvenclech committed:
refactor(routing): Chapters & Courses, add Pages

Princesseuh committed:
style: Format every file

Princesseuh committed:
feat(shiki): Add Ren'py language definition

Princesseuh committed:
fix(content): Fix some syntax mistakes causing MDX to crash

Princesseuh committed:
feat(routing): Implement routing for courses, chapters and pages

Princesseuh committed:
config(prettier): Ignore MDX files and TextMate grammars

Princesseuh committed:
chore(shiki): Add license for Ren'Py's language definition

Princesseuh committed:
fix(pages): Fix getPagesForChapter returning pages in other chapters than the specified one

Princesseuh committed:
fix(glob): Add a safeguard around `getContentForPath` to prevent crashing the dev server

Princesseuh committed:
feat(frontmatter): Add `url` property to BaseFrontmatter

Princesseuh committed:
refactor(pages): Remove unintended plural in `getPagesForChapter`

Princesseuh committed:
feat(nav): Add left navigation menu to navigate throughout a course's chapters and their pages

Princesseuh committed:
fix(content): Fix wrongly named file in itchio course

aureliendossantos committed:
feat(content): Update contribution docs for Astro

Princesseuh committed:
feat(content): Setup autoimports of components commonly used in content

Princesseuh committed:
feat(assets): Add Image and Video components for usage of respective assets inside content

Princesseuh committed:
feat(remark): Add Remark plugin to allow for relative links to assets inside Markdown

Princesseuh committed:
fix(pico-8): Fix chapter.mdx with wrong file extension

Princesseuh committed:
feat(style): Add pixelated class for pixelated image rendering

aureliendossantos committed:
feat(content): Use Image component everywhere

Princesseuh committed:
fix(content): Fix wrong file extension for chapter.mdx files in Pico-8 course

Princesseuh committed:
fix(build): Add workaround to make images work in build

Princesseuh committed:
fix(build): Fix image path in copy script

Princesseuh committed:
fix(avatars): Add missing assets for bubbles

aureliendossantos committed:
feat(content): Use Video component everywhere

Princesseuh committed:
feat(video): Add support for posters

aureliendossantos committed:
feat(content): YouTube component

aureliendossantos committed:
fix(video): Wrong caption

Princesseuh committed:
fix(assets): Generate hashed assets for Video and Posters to avoid collisions between assets with same names

Princesseuh committed:
chore(deps): Upgrade deps

Princesseuh committed:
fix(build): Fix videos not working in build

Princesseuh committed:
perf(netlify): Add cache headers

Princesseuh committed:
chore: remove unnecessary file

Princesseuh committed:
perf(prefetch): Add quicklink for prefetching links in viewport

aureliendossantos committed:
feat(content): New chapter 'Scripting in Ruby' in RPG Maker

Chapter ported from the wiki with minor adjustments
